packages/lisk-transactions/src/4_multisignature_transaction.ts

```diff
@@ -331,4 +332,65 @@ export class MultisignatureTransaction extends BaseTransaction {
 
 		return [];
 	}
+
+	public addMultisignature(
+		store: StateStore,
+		signatureObject: SignatureObject,
+	): TransactionResponse {
+		// Get the account
+		const account = store.account.get(this.senderId);
+		// Check if signature is not already there
+		if (this.signatures.includes(signatureObject.signature)) {
+			return createResponse(this.id, [
+				new TransactionError(
+					`Signature '${
+						signatureObject.signature
+					}' already present in transaction.`,
+					this.id,
+				),
+			]);
+		}
+
+		// Validate signature key belongs to pending multisig registration transaction
+		// tslint:disable-next-line
+		const keysgroup = (this as any).asset.multisignature.keysgroup.map(
+			(aKey: string) => aKey.slice(1),
+		);
+
+		if (!keysgroup.includes(signatureObject.publicKey)) {
+			return createResponse(this.id, [
+				new TransactionError(
+					`Public Key '${
+						signatureObject.publicKey
+					}' is not a member for account '${account.address}'.`,
+					this.id,
+				),
+			]);
+		}
+
+		// Validate the signature using the signature sender and transaction details
+		const trsSignature = validateMultisignatures(
+			[signatureObject.publicKey],
+			[signatureObject.signature],
+			1,
+			this.getBasicBytes(),
+		);
+		// If the signature is valid for the sender push it to the signatures array
+		if (trsSignature.valid) {
```

Comment:
need to check `valid || (errors.length > 0 && instanceof errors[0] === TransactionPendingError)`.
maybe 
```
if (!valid && instanceof errors[0] !== TransactionPendingError) {
 return createResponse(this.id, errors)
}
this.signatures.push(signatureObject.signature);
return this.processMultisignatures();
```